features/networking/operator.feature

```diff
@@ -13,3 +13,167 @@ Feature: Operator related networking scenarios
+      | resource_name | cluster                         |
+      | p             | {"spec":{"networkType":"None"}} |
+      | type          | merge                           |
+    Then the step should succeed
+    Given I register clean-up steps:
+    """
+    When I run the :patch admin command with:
+      | resource      | network.config.openshift.io                       |
+      | resource_name | cluster                                           |
+      | p             | {"spec":{"networkType":"<%= cb.network_type %>"}} |
+      | type          | merge                                             |
+    Then the step should succeed
+    """
+    # Check that the operator status reflect the problem
+    Given I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | clusteroperators                                                  |
+      | resource_name | network                                                           |
+      | o             | jsonpath='{.status.conditions[?(.type == "Failing")].status}'     |
+    Then the step should succeed
+    And the output should contain "True"
+    When I run the :get admin command with:
+      | resource      | clusteroperators                                                  |
+      | resource_name | network                                                           |
+      | o             | jsonpath='{.status.conditions[?(.type == "Available")].status}'   |
+    Then the step should succeed
+    And the output should contain "True"
+    """
+    # Change the network.config.openshift.io back
+    When I run the :patch admin command with:
+      | resource      | network.config.openshift.io                       |
+      | resource_name | cluster                                           |
+      | p             | {"spec":{"networkType":"<%= cb.network_type %>"}} |
+      | type          | merge                                             |
+    Then the step should succeed
+    # Check that the operator status
+    Given I wait up to 20 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | clusteroperators                                                  |
+      | resource_name | network                                                           |
+      | o             | jsonpath='{.status.conditions[?(.type == "Failing")].status}'     |
+    Then the step should succeed
+    And the output should contain "False"
+    When I run the :get admin command with:
+      | resource      | clusteroperators                                                  |
+      | resource_name | network                                                           |
+      | o             | jsonpath='{.status.conditions[?(.type == "Available")].status}'   |
+    Then the step should succeed
+    And the output should contain "True"
+    """
```

Comment:
worth writing a step `cluster network is ready` or something like this

Reply:
Added a new step to check the status for network operator with different condition types